src/main/groovy/com/twcable/grabbit/jcr/JcrNodeDecorator.groovy

```diff
@@ -45,11 +49,41 @@ class JcrNodeDecorator {
     private Collection<JcrNodeDecorator> immediateChildNodes
 
 
-    JcrNodeDecorator(@Nonnull JCRNode node) {
+    JcrNodeDecorator(@Nonnull final JCRNode node) {
         if(!node) throw new IllegalArgumentException("node must not be null!")
         this.innerNode = node
     }
 
+    static JcrNodeDecorator createFromProtoNode(@Nonnull final ProtoNodeDecorator protoNode, @Nonnull final Session session) {
+        final JcrNodeDecorator theDecorator = new JcrNodeDecorator(getOrCreateNode(protoNode, session))
+        final mandatoryNodes = protoNode.getMandatoryChildNodes()
+        mandatoryNodes?.each {
+            it.writeToJcr(session)
+        }
+        //If a version exception is thrown,
+        try {
+            protoNode.writableProperties.each { it.writeToNode(innerNode) }
```

Comment:
@jbornemann : I have merged your commit to this... Now we are adding mixins along with other properties whereas earlier we were adding mixins first and then other props... and there was a comment for the same:
-        //Write mixin types first to avoid InvalidConstraintExceptions
Just wanted to confirm is this will not be an issue before changing it again